### ISSUE TITLE
[DataGrid] Keyboard navigation only work in the first few rows

### ISSUE THREAD

**Issue by oliviertassinari** [mui-org/material-ui-x#1822]
- [x] The issue is present in the latest release.
- [x] I have searched the [issues](https://github.com/mui-org/material-ui-x/issues) of this repository and believe that this is not a duplicate.

## Current Behavior 😯

Keyboard navigation only work in the first few rows

## Expected Behavior 🤔

Keyboard navigation should work everywhere

## Steps to Reproduce 🕹

Steps:

1. Open https://codesandbox.io/s/1l40y
2.

![keyboard](https://user-images.githubusercontent.com/3165635/120666929-48751e80-c48d-11eb-98b6-7426a4a300f1.gif)

## Context 🔦

Raised in https://github.com/mui-org/material-ui-x/issues/1818#issuecomment-853770581

## Your Environment 🌎

v4.0.0-alpha.30

**Comment by finnhvman** [comment]
Is there maybe any workaround for this?

I also found that the Keyboard Navigation doesn't work when the rows are filtered.

**Comment by oliviertassinari** [comment]
@finnhvman Follow #1911, it should be fixed for free with this one.
